src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationOptions.cs

```diff
@@ -27,41 +31,121 @@ public LinkedInAuthenticationOptions()
             AuthorizationEndpoint = LinkedInAuthenticationDefaults.AuthorizationEndpoint;
             TokenEndpoint = LinkedInAuthenticationDefaults.TokenEndpoint;
             UserInformationEndpoint = LinkedInAuthenticationDefaults.UserInformationEndpoint;
+            EmailAddressEndpoint = LinkedInAuthenticationDefaults.EmailAddressEndpoint;
 
-            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "id");
-            ClaimActions.MapJsonKey(ClaimTypes.Email, "emailAddress");
-            ClaimActions.MapJsonKey(ClaimTypes.Name, "formattedName");
-            ClaimActions.MapJsonKey(ClaimTypes.GivenName, "firstName");
-            ClaimActions.MapJsonKey(ClaimTypes.Surname, "lastName");
-            ClaimActions.MapJsonKey(Claims.MaidenName, "maidenName");
-            ClaimActions.MapJsonKey(Claims.ProfileUrl, "publicProfileUrl");
-            ClaimActions.MapJsonKey(Claims.PictureUrl, "pictureUrl");
-            ClaimActions.MapJsonKey(Claims.Industry, "industry");
-            ClaimActions.MapJsonKey(Claims.Summary, "summary");
-            ClaimActions.MapJsonKey(Claims.Headline, "headline");
-            ClaimActions.MapCustomJson(Claims.Positions, user => user["positions"]?.ToString());
-            ClaimActions.MapJsonKey(Claims.PhoneticFirstName, "phoneticFirstName");
-            ClaimActions.MapJsonKey(Claims.PhoneticLastName, "phoneticLastName");
-            ClaimActions.MapJsonKey(Claims.FormattedPhoneticName, "formattedPhoneticName");
-            ClaimActions.MapCustomJson(Claims.Location, user => user["location"]?.ToString());
-            ClaimActions.MapJsonKey(Claims.Specialties, "specialties");
-            ClaimActions.MapJsonKey(Claims.NumConnections, "numConnections");
-            ClaimActions.MapJsonKey(Claims.NumConnectionsCapped, "numConnectionsCapped");
-            ClaimActions.MapJsonKey(Claims.CurrentShare, "currentShare");
-            ClaimActions.MapCustomJson(Claims.PictureUrls, user => user["pictureUrls"]?.ToString());
+            Scope.Add("r_liteprofile");
+            Scope.Add("r_emailaddress");
+
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, ProfileFields.Id);
+            ClaimActions.MapJsonKey(ClaimTypes.Email, LinkedInAuthenticationConstants.EmailAddressField);
+            ClaimActions.MapCustomJson(ClaimTypes.Name, user => GetFullName(user));
+            ClaimActions.MapCustomJson(ClaimTypes.GivenName, user => GetMultiLocaleString(user, ProfileFields.FirstName));
+            ClaimActions.MapCustomJson(ClaimTypes.Surname, user => GetMultiLocaleString(user, ProfileFields.LastName));
+            ClaimActions.MapCustomJson(Claims.PictureUrl, user => GetPictureUrls(user)?.LastOrDefault());
+            ClaimActions.MapCustomJson(Claims.PictureUrls, user => string.Join(",", GetPictureUrls(user)));
```

Comment:
As noted here https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/issues/300#issuecomment-494873451 this needs to handle there being no pictures.

Reply:
Yes, just noticed with my unit tests, update is on its way!